sky/utils/vpn_utils.py

```
@@ -0,0 +1,285 @@
+        return {}
+
+    def get_private_ip(self, hostname: str) -> str:
+        """Get the private IP address from the hostname."""
+        raise NotImplementedError
+
+    def remove_host(self, hostname: str) -> None:
+        """Remove a host from the VPN."""
+        raise NotImplementedError
+
+    def to_yaml_config(self) -> Dict[str, Any]:
+        """Get the VPN configuration in YAML format."""
+        return {}
+
+    @staticmethod
+    def from_backend_config(config: Dict[str, Any]) -> 'VPN':
+        """Create a VPN configuration from the backend configuration."""
+        raise NotImplementedError
+
+    def to_backend_config(self) -> Dict[str, Any]:
+        """Get the VPN configuration for the backend."""
+        return {}
+
+
+class TailscaleVPN(VPN):
+    """Tailscale VPN configuration."""
+
+    # pylint: disable=line-too-long
+    _SETUP_COMMAND = (
+        'curl -fsSL https://pkgs.tailscale.com/stable/ubuntu/focal.noarmor.gpg | '
+        'sudo tee /usr/share/keyrings/tailscale-archive-keyring.gpg >/dev/null; '
+        'curl -fsSL https://pkgs.tailscale.com/stable/ubuntu/focal.tailscale-keyring.list | '
+        'sudo tee /etc/apt/sources.list.d/tailscale.list >/dev/null; '
+        'sudo apt-get update > /dev/null 2>&1; '
+        'sudo apt-get install tailscale -y > /dev/null 2>&1; '
+        'sudo tailscale login --auth-key {tailscale_auth_key}')
+
+    def __init__(
+        self,
+        auth_key: str,
+        api_key: Optional[str] = None,
+        network_name: Optional[str] = None,
+        enable_api: bool = False,
+    ) -> None:
+        self._auth_key = auth_key
+        self._api_key = api_key
+        self._network_name = network_name
+        self._enable_api = enable_api
+        if not enable_api:
+            logger.warning('TAILSCALE_API_KEY or TAILSCALE_NETWORK_NAME is'
+                           ' not set. Tailscale API will be disabled. You'
+                           ' may need to remove hosts manually.')
```

Lets raise an error if we find the api is not enabled